### PR TITLE
Nullable enums

### LINKED ISSUE #19: Support for nullable enums
`PlatenumDoctrineType` does not support nullable fields. When entity which uses enums on nullable fields is persisted, there is `Call to a member function getValue() on null` error. Unfortunately since class is `final` it's impossible to override `convertToDatabaseValue()` and `convertToPHPValue()` methods.

### PR DESCRIPTION
**Pull request by Wirone**
Fixes #19.

### CODE REVIEW

**Comment by Wirone** [comment]
CI fails because of Psalm, but errors are not from my changes. I've suppressed one and fixed the other, but it failed again - due to not-existing `composer.lock` different versions of `doctrine/dbal` are installed and `Type::getSQLDeclaration()`'s signature differs between them and it's better to suppress this error too, what I did in the end.

File: src/Doctrine/PlatenumDoctrineType.php [hunk 1]
```diff
@@ -101,19 +101,30 @@ public function getName(): string
         return $this->platenumAlias;
     }
 
-    public function getSQLDeclaration(array $declaration, AbstractPlatform $platform): string
+    /**
+     * @psalm-suppress ParamNameMismatch
```
Comment (thunderer) [inline src/Doctrine/PlatenumDoctrineType.php]: please one-line this docblock

File: src/Doctrine/PlatenumDoctrineType.php [hunk 2]
```diff
@@ -101,19 +101,30 @@ public function getName(): string
...
+     */
+    public function getSQLDeclaration(array $column, AbstractPlatform $platform): string
     {
-        return ($this->platenumSql)($declaration, $platform);
+        return ($this->platenumSql)($column, $platform);
     }
 
     public function convertToDatabaseValue($value, AbstractPlatform $platform)
     {
+        if (null === $value || '' === $value) {
```
Comment (thunderer) [inline src/Doctrine/PlatenumDoctrineType.php]: please remove the empty string condition, only `null` should be allowed here

File: src/Doctrine/PlatenumDoctrineType.php [hunk 3]
```diff
@@ -101,19 +101,30 @@ public function getName(): string
...
+            return null;
+        }
+
         /** @psalm-suppress MixedMethodCall */
         return ($this->platenumCallback)($value->getValue());
     }
 
     public function convertToPHPValue($value, AbstractPlatform $platform)
     {
+        if (null === $value || '' === $value) {
```
Comment (thunderer) [inline src/Doctrine/PlatenumDoctrineType.php]: please remove the empty string condition, only `null` should be allowed here

**Review by thunderer — CHANGES_REQUESTED** [review]
@Wirone thank you for the contribution, please apply my suggestions and I'll merge this PR. It would be beneficial to also squash the commits and write a descriptive commit message.

**Comment by thunderer** [comment]
@Wirone as for that Psalm `ParamNameMismatch` issue - the parameter name changed in previous Doctrine DBAL versions and Psalm now checks that because of PHP8 [named parameters RFC](https://wiki.php.net/rfc/named_params). As we run multiple PHP versions in Travis, this will obviously fail for some jobs.

File: src/Doctrine/PlatenumDoctrineType.php (hunk 1 above)
Comment (Wirone) [reply src/Doctrine/PlatenumDoctrineType.php]: I am used to add one-line docblocks above properties or statements, for classes/methods I always use multiline. But I changed it as requested.

File: src/Doctrine/PlatenumDoctrineType.php (hunk 2 above)
Comment (Wirone) [reply src/Doctrine/PlatenumDoctrineType.php]: Done

File: src/Doctrine/PlatenumDoctrineType.php (hunk 3 above)
Comment (Wirone) [reply src/Doctrine/PlatenumDoctrineType.php]: Done

**Review by thunderer — APPROVED** [review]

**Comment by thunderer** [comment]
@Wirone I just released `v0.3.1` including your PR. Thanks again for your contribution and using Platenum. Let me know if you have any more questions.

**Pull request merged**